src/ORM/Table.php

```diff
@@ -922,6 +922,17 @@ public function behaviors()
         return $this->_behaviors;
     }
 
+    /**
+     * Get a behavior from the registry.
+     *
+     * @param string $name The behavior alias to get from the registry.
+     * @return \Cake\ORM\Behavior|null
```

Comment:
A getter ideally doesn't return null, it throws a clear and meaningful exception instead.
Especially since a "has" method is available.

This makes code flow easier, as you can expect the object to be there to directly use the return value in chaining.
The get vs find pattern we use everywhere else, as well, where possible.

Reply:
`$this->behaviors()->get('BehaviorName')` returns `null` ...

Reply:
Even that one shouldnt return null :) But that is now in code, and we need to wait for 4.x to fix that for BC reasons.
For now we shouldn't introduce more baggage then to clean up with the next major IMO.

Reply:
I'd be happy to change it if you'd like @dereuromark. I added the `null` because that is what the `ObjectRegistry` had. Would you like me to update the docblock?

Reply:
Also, should the test be updated to not check for a `null` return?

Reply:
Let's see what the discussion around my proposal https://github.com/cakephp/cakephp/pull/11638 yields first.

Reply:
So, yeah, you can remove the nullable part and throw an exception instead. People can check with hasBehavior() if needed.